spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -608,4 +693,23 @@ private DatastoreReaderWriter getDatastoreReadWriter() {
 			throw new DatastoreDataException("Property matchers are not supported");
 		}
 	}
+
+	/**
+	 * Class to hold caches for read and conversion.
+	 *
+	 * @author Dmitry Solomakha
+	 */
+	class ReadContext {
```

Comment:
Since you already have this class, why not make it a bit smarter?
For example:
```java
Object getConvertedEntity(Key key)
```
The method would automatically convert if needed, or return `null`.